src/vs/workbench/contrib/issue/browser/issueQuickAccess.ts

```diff
@@ -0,0 +1,156 @@
+
+	static PREFIX = 'issue ';
+
+	constructor(
+		@IMenuService private readonly menuService: IMenuService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService,
+		@ICommandService private readonly commandService: ICommandService,
+		@IExtensionService private readonly extensionService: IExtensionService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IProductService private readonly productService: IProductService
+	) {
+		super(IssueQuickAccess.PREFIX, { canAcceptInBackground: true });
+	}
+
+	protected override _getPicks(filter: string): Picks<IPickerQuickAccessItem> | FastAndSlowPicks<IPickerQuickAccessItem> | Promise<Picks<IPickerQuickAccessItem> | FastAndSlowPicks<IPickerQuickAccessItem>> | null {
+		if (this.configurationService.getValue<string>('extensions.experimental.quickAcessExtensions')) {
+			const issuePicks: Array<IPickerQuickAccessItem | IQuickPickSeparator> = [];
+			const extensionIdList: Array<string> = [];
+
+			// add regular open issue reporter button
+			const createTerminalLabel1 = this.productService.quality === 'stable' ? localize("workbench.action.openIssueReporter", "Visual Studio Code") : localize("workbench.action.openIssueReporterInsiders", "Visual Studio Code: Insiders");
+			issuePicks.push({
+				label: `$(plus) ${createTerminalLabel1}`,
+				ariaLabel: createTerminalLabel1,
+				accept: () => this.commandService.executeCommand('workbench.action.openIssueReporter', { issueSource: IssueSource.VSCode })
+			});
+
+			if (issuePicks.length > 0) {
+				issuePicks.push({ type: 'separator' });
+			}
+
+			const createTerminalLabel2 = localize("workbench.action.openIssueReporter2", "Extension Marketplace");
+			issuePicks.push({
+				label: `$(plus) ${createTerminalLabel2}`,
+				ariaLabel: createTerminalLabel2,
+				accept: () => this.commandService.executeCommand('workbench.action.openIssueReporter', { issueSource: IssueSource.Marketplace })
+			});
+
+
+
+			if (issuePicks.length > 0) {
+				issuePicks.push({ type: 'separator', label: localize('extensions', "Extensions: Custom Reporting") });
+			}
+			// creates menu from contributed
+			const menu = this.menuService.createMenu(MenuId.IssueReporter, this.contextKeyService);
+
+			// render menu and dispose
+			const actions = menu.getActions({ renderShortTitle: true }).flatMap(entry => entry[1]);
+
+
+			// create picks from contributed menu
+			actions.forEach(action => {
+				if ('source' in action.item && action.item.source) {
+					extensionIdList.push(action.item?.source?.id);
+				}
+
+				const pick = this._createPick(filter, action);
+				if (pick) {
+					issuePicks.push(pick);
+				}
+			});
+
+			if (issuePicks.length > 0) {
+				issuePicks.push({ type: 'separator', label: localize('otherExtensions', "Other Extensions") });
+			}
+
+			// create picks from extensions
+			this.extensionService.extensions.forEach(extension => {
+				if (!extension.isBuiltin) {
```

Comment:
Is there a reason you are doing the menu stuff? If the label is just the extension name, couldn't you just get it from `this.extensionService.extensions` with the rest of the extensions that don't implement the menu?

Reply:
the main reason I have to do this to trigger the contributed action is because calling `this.commandService.executeCommand('workbench.action.openIssueReporter'` with extension id won't trigger the command actually. the command is triggered and checked thru the original flow (help -> extension -> select extension). so it's triggered on that menu `change` instead of on open.